### PR TITLE
feat: deprecate mongo db like QL syntax in redis and support native QL

### PR DESCRIPTION
**Pull request by alaeddine-13**
Support native Redis QL and deprecate mongo DB like QL in Redis filtering.

- [x] Redis filtering
- [x] docs
- [x] tests

closes: https://github.com/jina-ai/docarray/issues/639

### CODE REVIEW

**Review by JoanFM — APPROVED** [review]
Let's wait for @NicholasDunham input, but looks good to me

**Review by JohannesMessner — APPROVED** [review]
LGTM

File: docs/advanced/document-store/redis.md [hunk 1]
```diff
@@ -113,17 +113,41 @@ da2.summary()
...
+| `ef_runtime`      | Optional parameter for Redis HNSW algorithm                                                       | `10`                                              |
+| `block_size`      | Optional parameter for Redis FLAT algorithm                                                       | `1048576`                                         |
+| `initial_cap`     | Optional parameter for Redis HNSW and FLAT algorithm                                              | `None`, defaults to the default value in Redis    |
+| `columns`         | Other fields to store in Document and build schema                                                | `None`                                            |
 
-You can perform Vector Similarity Search based on [FLAT or HNSW algorithm](vector-search-index) and pre-filter results using a filter query that is based on [MongoDB's Query](https://www.mongodb.com/docs/manual/reference/operator/query/). The following tags filters can be combine with `$and` and `$or`:
+You can check the default values in [the docarray source code](https://github.com/jina-ai/docarray/blob/main/docarray/array/storage/redis/backend.py)
```
Comment (NicholasDunham) [inline docs/advanced/document-store/redis.md:140]:
```suggestion
You can check the default values in [the docarray source code](https://github.com/jina-ai/docarray/blob/main/docarray/array/storage/redis/backend.py).
```

File: docs/advanced/document-store/redis.md [hunk 2]
```diff
@@ -113,17 +113,41 @@ da2.summary()
...
+For vector search configurations, default values are those of the database backend, you can find them in [redis documentation](https://redis.io/docs/stack/search/reference/vectors/)
```
Comment (NicholasDunham) [inline docs/advanced/document-store/redis.md:141]:
```suggestion
For vector search configurations, default values are those of the database backend, which you can find in the [Redis documentation](https://redis.io/docs/stack/search/reference/vectors/).
```

File: docs/advanced/document-store/redis.md [hunk 3]
```diff
@@ -225,49 +230,73 @@ This will print:
...
+````{admonition} Note
+:class: note
+Note that Redis does not support Boolean types in attributes. Therefore, you need to configure your boolean field as 
+integer in `columns` configuration (`'field': 'int'`) and use a filter query that treats the field as an integer
+(`@field: [1 1]`).
+````
+
+### Search by filter query
+
+One can search with user-defined query filters using the `.find` method. Such queries follow the [Redis' Search Query Syntax](https://redis.io/docs/stack/search/reference/query_syntax/).
```
Comment (NicholasDunham) [inline docs/advanced/document-store/redis.md:248]:
```suggestion
One can search with user-defined query filters using the `.find` method. Such queries follow the [Redis Search Query Syntax](https://redis.io/docs/stack/search/reference/query_syntax/).
```

File: docs/advanced/document-store/redis.md [hunk 4]
```diff
@@ -225,49 +230,73 @@ This will print:
...
+
+Consider you store Documents with a certain tag `price` into Redis and you want to retrieve all Documents
```
Comment (NicholasDunham) [inline docs/advanced/document-store/redis.md:250]:
```suggestion
Consider a case where you store Documents with a tag of `price` into Redis and you want to retrieve all Documents
```

File: docs/advanced/document-store/redis.md [hunk 5]
```diff
@@ -225,49 +230,73 @@ This will print:
...
+with `price`  lower or equal to  some `max_price` value.
```
Comment (NicholasDunham) [inline docs/advanced/document-store/redis.md:251]:
```suggestion
with `price` less than or equal to  some `max_price` value.
```

File: docs/advanced/document-store/redis.md [hunk 6]
```diff
@@ -225,49 +230,73 @@ This will print:
...
-{
-    "$or": {
-        "brand": {"$eq": "Nike"},
-        "$and": {
-            "price": {"$lt": 100},
-            "color": {"$eq": "blue"},
-        },
-    }
-}
+Then you can retrieve all documents whose price is lower than or equal to `max_price` by applying the following
```
Comment (NicholasDunham) [inline docs/advanced/document-store/redis.md:275]:
```suggestion
Then you can retrieve all documents whose price is less than or equal to `max_price` by applying the following
```

**Review by NicholasDunham — CHANGES_REQUESTED** [review]
Minor grammatical changes

**Review by NicholasDunham — APPROVED** [review]
LGTM

**Pull request merged**